app/controllers/articles_controller.rb

```diff
@@ -0,0 +1,53 @@
+class ArticlesController < ApplicationController
+  
+  http_basic_authenticate_with name: "user", password: "secret", except: [:index, :show]
+
+  def index
+    @articles = Article.all
+  end
+
+  def show
+    @article = Article.find(params[:id])
+  end
+
+  def new
+    @article = Article.new
+  end
+
+  def create
+    # @article = Article.new(title: "Amazing Animals", body: "Show of top 10 Animals"title: "Amazing Animals", body: "Show of top 10 Animals")
+    @article = Article.new(article_params)
+    if @article.save
+      redirect_to(article_path(@article))
+    else
+      render :new
+    end
+  end
+
+  def edit
+    @article = Article.find(params[:id])
+  end
+
+  def update
+    @article = Article.find(params[:id])
+
+    if @article.update(article_params)
+      redirect_to @article
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    @article = Article.find(params[:id])
```

Comment:
You are fetching the article in several actions `Article.find(params[:id])`. What can you do to avoid repetition in that case?

Reply:
Create a `Concern` for controllers which return the object if it was found. I will create the concern

Reply:
Why a `Concern`?

Reply:
Because `Article.find(params[:id])` is used in comments_controller many times too, but the other solution is to make a method that returns the found Article. I think creating a Concern for these controllers is a good solution.

Reply:
```
module Visible
  extend ActiveSupport::Concern

  def find(id)
    Article.find(id)
  end
end
```

Reply:
Final solution
```
before_action :find_article, except: %i[index show]
.....
private
def find_article
    @article = Article.find(params[:id])
end
```